grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -333,32 +334,33 @@ class CycleCountService {
                     facility,
                     availableItemToRecount,
                     cycleCount,
-                    command.countIndex,
+                    countIndex,
                     CycleCountItemStatus.INVESTIGATING)
 
             cycleCount.addToCycleCountItems(cycleCountItem)
         }
 
-        // If there are custom items we need to create adjusted non-custom items with the appropriate
-        // cycle count index
-        for (CycleCountItem customCycleCountItem : cycleCount.cycleCountItems) {
+        // If there are custom items, they won't have been discovered by the above loop (because they won't have
+        // a product availability record) so make sure to create new recount items for them as well.
+        // "Most recent count" in this scenario is the previous count.
+        Set<CycleCountItem> customItemsOfLastCount = cycleCount.itemsOfMostRecentCount.findAll{ it.custom }
+        for (CycleCountItem customCycleCountItem : customItemsOfLastCount) {
```

Comment:
This is the real fix. Don't loop the same list that we add to. Everything else is just formatting or variable renames